### DISCUSSION TITLE
Resolve the types of a method parameters

### DISCUSSION THREAD

**Discussion by kironia** [kironia/99balloons#2] [Q&A] [answered]
Given a Java method with two parameters of type int and a second method with two parameters of type int and long, when I call them passing 41 and 63 as their arguments, which one is called? Of course, those two methods are named the same and are in the same class, thus raising a question about that ambiguity.

**Comment by Stagyrite** [comment, marked as answer]
Let's assume that the methods are named 'typeOf'. Thus, candidates for method call typeOf(41, 63) are:

- void typeOf(int, int)
- void typeOf(int, long)

The right method call is the one of type 'void typeOf(int, int)'. Let's assume that the methods should write either 'Integer' or 'Long' depending on the second argument type. Here's what an example class should look like.

```
class TypeDetector {

    static void typeOf(int numberX, int numberY) {
        System.out.println("Integer");
    }

    static void typeOf(int numberX, long numberY) {
        System.out.println("Long");
    }

}
```
The output is always 'Integer', as one can try in JShell.
```
|  Welcome to JShell -- Version 17.0.10
|  For an introduction type: /help intro

jshell> class TypeDetector {
   ...>
   ...>     static void typeOf(int numberX, int numberY) {
   ...>         System.out.println("Integer");
   ...>     }
   ...>
   ...>     static void typeOf(int numberX, long numberY) {
   ...>         System.out.println("Long");
   ...>     }
   ...>
   ...> }
   ...>
|  created class TypeDetector

jshell> TypeDetector.typeOf(41, 63);
Integer
```
